### PR TITLE
Logout

### LINKED ISSUE #405: logout
logout function is missing

### PR DESCRIPTION
**Pull request by audax**
Logout function which only deletes the token and only allows a login with the same account name and email.

Resolves #405

### CODE REVIEW

**Comment by audax** [comment]
retest this please

**Comment by julianseeger** [comment]
whats up with this PR? assigned to me and wip?

**Comment by audax** [comment]
You were assigned before I reapplied the WIP label…

I don't exactly know why the tests are failing atm.

**Comment by audax** [comment]
The BoxProviderTests are not relevant to this PR and are probably failing because the internet connection is bad oder the test-server is bad or whatever is bad. They should be run with a mock instead.

@julianseeger please pretend that those tests are not failing.

File: qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/LogoutUITest.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        appPreference.setAccountEMail(ACCOUNT_E_MAIL);
+    }
+
+    @Test
+	public void testLogout() {
+        setAccountPreferences();
+        appPreference.setToken(TestConstants.TOKEN);
+        Intents.release();
+        mainActivityActivityTestRule.launchActivity(null);
+		openDrawer(R.id.drawer_layout);
```
Comment (julianseeger) [inline qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/LogoutUITest.java]: ;'(

File: qabelbox/src/main/java/de/qabel/qabelbox/activities/MainActivity.java [hunk 2]
```diff
@@ -782,6 +791,8 @@ public boolean onNavigationItemSelected(MenuItem item) {
             selectAboutFragment();
         } else if (id == R.id.nav_help) {
             selectHelpFragment();
+        } else if (id == R.id.nav_logout) {
+			performLogout();
```
Comment (julianseeger) [inline qabelbox/src/main/java/de/qabel/qabelbox/activities/MainActivity.java]: ;'(

File: qabelbox/src/main/java/de/qabel/qabelbox/fragments/CreateAccountLoginFragment.java [hunk 3]
```diff
@@ -33,6 +33,14 @@
 
     private BoxAccountRegisterServer mBoxAccountServer;
     private View resetPassword;
+    private String account_email;
```
Comment (julianseeger) [inline qabelbox/src/main/java/de/qabel/qabelbox/fragments/CreateAccountLoginFragment.java]: LCC please

File: qabelbox/src/main/java/de/qabel/qabelbox/fragments/CreateAccountMainFragment.java [hunk 4]
```diff
@@ -9,23 +9,37 @@
...
- */
 public class CreateAccountMainFragment extends BaseIdentityFragment implements View.OnClickListener {
 
     private Button mCreateAccount;
     private Button mLogin;
+    private boolean skipToLogin = false;
+
+    public static String SKIP_TO_LOGIN = "SKIP_TO_LOGIN";
+    private String account_name;
+    private String account_email;
```
Comment (julianseeger) [inline qabelbox/src/main/java/de/qabel/qabelbox/fragments/CreateAccountMainFragment.java]: same here

File: qabelbox/src/main/java/de/qabel/qabelbox/fragments/CreateAccountResetPasswordFragment.java [hunk 5]
```diff
@@ -28,6 +28,12 @@
 
     private TextView etEMail;
     private BoxAccountRegisterServer mBoxAccountServer;
+    private String account_email;
```
Comment (julianseeger) [inline qabelbox/src/main/java/de/qabel/qabelbox/fragments/CreateAccountResetPasswordFragment.java]: and here

File: qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/LogoutUITest.java [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        service = uiBoxHelper.getService();
+        uiBoxHelper.createTokenIfNeeded(false);
+        uiBoxHelper.removeAllIdentities();
+        uiBoxHelper.addIdentity("identity");
+    }
+
+    @After
+    public void tearDown() {
+        if (appPreference != null) {
+            // Because other tests are probably not correctly isolated
```
Comment (julianseeger) [inline qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/LogoutUITest.java]: maybe rename `tearDown` to `clearLeftoverOfBadlyIsolatedTests`?

File: qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/LogoutUITest.java [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    public static final String ACCOUNT_NAME = "account_name";
+    public static final String ACCOUNT_E_MAIL = "accountmail@example.com";
+    IntentsTestRule<MainActivity> mainActivityActivityTestRule =
+            new MainActivityWithoutFilesFragmentTestRule();
+    private AppPreference appPreference;
+    private LocalQabelService service;
+
+
+    @Before
+    public void setUp() {
```
Comment (julianseeger) [inline qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/LogoutUITest.java]: setUp and tearDown of LoginUiTest and LogoutUiTest looks duplicated. If so, you could merge them to some AccountingUiTest or similar

Comment (audax) [inline qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/LogoutUITest.java]: The problem with that are the different test rules, but I could extract a super class.

**Comment by julianseeger** [comment]
So what about the other failing tests (those not containing `Box` in their name)?

**Comment by audax** [comment]
Either my last commit broke half of the application or the test-block server is down.

**Comment by julianseeger** [comment]
Even a `ChatMessageUiTest` is broken ?!

**Comment by audax** [comment]
Yes, because many, many tests wait until a BoxVolume was created before they then manually navigate the app until the arrive in the fragment that they want to test. The ChatMessageUITest gets stuck waiting for the BoxVolume.

**Pull request merged**